fastestimator/op/tensorop/meta/one_of.py

```diff
@@ -48,22 +48,22 @@ def __init__(self, *tensor_ops: TensorOp) -> None:
             assert self.out_list == op.out_list, "All ops within OneOf must share the same output configuration"
             assert mode == op.mode, "All ops within a OneOf must share the same mode"
             assert ds_id == op.ds_id, "All ops within a OneOf must share the same ds_id"
+        if prob:
+            assert len(tensor_ops) == len(prob), "The number of probabilities do not match with number of Operators"
+            assert abs(sum(prob) - 1) < 1e-8, "Probabilities must sum to 1"
+        else:
+            prob = [1 / len(tensor_ops) for _ in tensor_ops]
         self.ops = tensor_ops
+        self.prob = prob
         self.prob_fn = None
```

Comment:
If you're moving away from prob_fn and invoke_fn then these need to be removed here, but they should be used in lieu of your new solution if possible in order to keep the forward method implementation consistent